siliconcompiler/core.py

```diff
@@ -513,40 +527,42 @@ def find_function(self, modulename, funcname, moduletype=None):
 
         '''
 
+        # Ensure that SC built-ins are on the $PYTHONPATH (only necessary for local installs)
+        if not self.scroot in sys.path:
+            sys.path.append(self.scroot)
+
         # module search path depends on modtype
+        module = None
         if moduletype is None:
             for item in ('targets', 'flows', 'tools', 'pdks', 'libs', 'checklists'):
-                fullpath = self.find_function(modulename, funcname, module_type=item)
-                if fullpath:
+                try:
+                    module = importlib.import_module(f'{item}.{modulename}')
                     break
+                except ModuleNotFoundError:
+                    pass
         elif moduletype in ('targets','flows', 'pdks', 'libs'):
-            fullpath = self._find_sc_file(f"{moduletype}/{modulename}.py", missing_ok=True)
+            try:
+                module = importlib.import_module(f'{moduletype}.{modulename}')
+            except ModuleNotFoundError:
+                pass
         elif moduletype in ('tools', 'checklists'):
-            fullpath = self._find_sc_file(f"{moduletype}/{modulename}/{modulename}.py", missing_ok=True)
+            modulefile = modulestep if modulestep is not None else modulename
+            try:
+                module = importlib.import_module(f'{moduletype}.{modulename}.{modulefile}')
+            except ModuleNotFoundError:
+                pass
```

Comment:
Might want to add a last check for just `import_module({modulename})` (or whatever is appropriate) since the other library might not be organized in the same tools/flows/etc?

Reply:
I looked into doing this, but I couldn't see a clean way to do it. We don't have any modules that are packaged in a top-level namespace, and it seems like namespacing could be hard. There's already a `klayout` package on PyPi, for example.